admin/modules/config/plugins.php

```diff
@@ -663,31 +665,47 @@ function build_plugin_list($plugin_list)
 			$uninstall_button = true;
 		}
 
-		$table->construct_cell("<strong>{$plugininfo['name']}</strong> ({$plugininfo['version']})<br /><small>{$plugininfo['description']}</small><br /><i><small>{$lang->created_by} {$plugininfo['author']}</small></i>");
+		if(function_exists($configure_func))
+		{
+			$configure_link = $configure_func();
+		}
+
+		$extra = array();
+		if ($installed != false && $configure_link == false) {
+			$extra['colspan'] = 2;
+		}
+
+		$table->construct_cell("<strong>{$plugininfo['name']}</strong> ({$plugininfo['version']})<br /><small>{$plugininfo['description']}</small><br /><i><small>{$lang->created_by} {$plugininfo['author']}</small></i>", $extra);
 
 		// Plugin is not installed at all
 		if($installed == false)
 		{
 			if($compatibility_warning)
 			{
-				$table->construct_cell("{$compatibility_warning}", array("class" => "align_center", "colspan" => 2));
+				$table->construct_cell("{$compatibility_warning}", array("class" => "align_center", "colspan" => 3));
 			}
 			else
 			{
-				$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=activate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\">{$lang->install_and_activate}</a>", array("class" => "align_center", "colspan" => 2));
+				$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=activate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\" class=\"button\">{$lang->install_and_activate}</a>", array("class" => "align_center", "colspan" => 3));
 			}
 		}
 		// Plugin is activated and installed
 		else if($plugininfo['is_active'])
 		{
-			$table->construct_cell("<a href=\"index.php?module=config-plugins&amp;action=deactivate&amp;plugin={$plugininfo['codename']}&amp;my_post_key={$mybb->post_code}\">{$lang->deactivate}</a>", array("class" => "align_center", "width" => 150));
+			if ($configure_link !== false) {
```

Comment:
Code style - this should have no space between `if` and the `(` and the `{` should be on a new line.

Reply:
Done in e3d47839eb1e6e066bdddc614b24e27f90f7816b 👻